roachtest: run cluster backup schedule by default on gce roachtests

msbutler opened the pull request:
This patch creates a cluster backup schedule by default on roachtests running on gce. A roachtest writer can override this default behavior by passing option.DefaultStartOptsNoBackups() at cluster.Start().

Roachtests are supposed to mirror a customer env. Like a customer's cluster, roachtests currently run background jobs like gc, sql stats computation, rebalancing etc. But unlike most customer clusters, roachtests do not run scheduled backup jobs by default. Any workload on a roachtest should run just as well with or without with background jobs, but we currently do not test this, at least wrt to scheduled backups. This patch addresses this lack of test coverage.

This new feature is disabled on any test that fails to create a backup schedule and all tests currently on roachperf, except for  `tpcc*`, `tpccbench*`, `tpch*`, `scbench`. Disabled tests include:
- kv0*
- decomissionbench*
- ycsb*
- admission-control*
- backup*
- restore*
- kvbench*

Fixes #86045

Release note: None

msbutler committed:
roachtest: run cluster backup schedule by default

This patch creates a cluster backup schedule by default on roachtests. A
roachtest writer can override this default behavior by passing
option.DefaultStartOptsNoBackups() at cluster.Start().

Roachtests are supposed to mirror a customer env. Like a customer's cluster,
roachtests currently run background jobs like gc, sql stats computation,
rebalancing etc. But unlike most customer clusters, roachtests do not run
scheduled backup jobs by default. Any workload on a roachtest should run just
as well with or without with background jobs, but we currently do not test
this, at least wrt to scheduled backups. This patch addresses this lack of test
coverage.

This new feature is disabled on performance sentive roachtests for which an
additional background job will make it harder to debug a perf regression. This
includes:
- kv0*
- decomissionbench*
- ycsb*
- admission-control*
- backup*
- restore*
- kvbench*

Fixes #86045

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/95460)
<!-- Reviewable:end -->

smg260 reviewed (commented):
Being able to more accurately mimic customer environments aligns with what roachtests are useful for. I think this is a reasonable enhancement, but it might be worth running this branch via TC roachtest nightly to see its impact (duration, errors etc)

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/95460#-:-NMAWBft-oVPdt3I_dHS:b-h4u9rr)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @dt and @herkolategan)



<!-- Sent from Reviewable.io -->

msbutler commented:
bors r=knz

srosenberg reviewed (commented):
Thanks for PR, Michael! My feedback is essentially the same as Miral's. It's a fairly large change wrt performance. Let's do one nightly run as a sanity check, primarily to confirm there are no unintended failures, or other noticeable side-effects. (Please ping #test-eng if you're unsure how to do this via TC.) Following the sanity check, we can merge and make sure we add an annotation to roachperf. If there are significant perf. regressions, that would be useful insight. Worst-case scenario, we may need to revert the change for some of the roachtests.

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/95460#-:-NMRdBqqCbml4tloYQRJ:byoxdf0)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @dt and @herkolategan)



<!-- Sent from Reviewable.io -->

msbutler committed:
roachprod: seperate syncedcluster.SQL() into c.ExecSQL() and
c.ExecOrInteractiveSQL()

Previously, a user of syncedCluster.SQL() could unknowingly cause a panic if
they called this function on a single node cluster, as this open an interactive
SQL shell. This patch prevents this footgun by seperating the c.SQL() call into
c.ExecSQL() and c.ExecOrInteractiveSQL().

Epic: none

Release note: None.

msbutler commented:
first legit [GCE nightly](https://teamcity.cockroachdb.com/viewType.html?buildTypeId=Cockroach_Nightlies_RoachtestNightlyGceBazel&tab=buildTypeStatusDiv&branch_Cockroach_Nightlies=95460) revealed a few tests should not run with scheduled backups. Going to run the nightly once more.

stevendanna commented on pkg/roachprod/install/cockroach.go:
```diff
@@ -221,7 +222,10 @@ func (c *SyncedCluster) Start(ctx context.Context, l *logger.Logger, startOpts S
+
+	// Only after a successful cluster initialization should we attempt to schedule backups.
+	if startOpts.ScheduleBackups && !startOpts.SkipInit {
+		c.Cluster.VMs.Len()
```
What's this for?

msbutler replied:
heh. nothing. a silly mistake. removing.

stevendanna commented on pkg/roachprod/roachprod.go:
```diff
@@ -435,7 +435,10 @@ func SQL(
 		return err
 	}
-	return c.SQL(ctx, l, tenantName, cmdArray)
+	if len(cmdArray) == 0 {
+		return c.OpenSQLShell(ctx, l, tenantName)
+	}
+	return c.RunSQL(ctx, l, cmdArray)
```
I'm boolean impaired today so perhaps I have just confused myself.  But does this end up minorly changing the behaviour of something like: `roachprod sql local:1 -- -u foo` where `foo` is a user in the cluster.  Previously this would open an interactive shell to the node. Now, I don't think it does.

msbutler replied:
huh, you're right. The behavior on master is: if one node is specified, ssh onto that node, and pass the cmd to ./cockroach sql. If many nodes are specified, try parallel execution. It's a confusing API, and I think I have over simplified it. I'll change this PR to retain the behavior, and will let test-eng ponder if this API behavior should change.

stevendanna reviewed (commented):
Thanks for getting to this. Left a few questions in the code.

I think that when this merges we should consider adding a roachperf annotation and also making some noise about it in the engineering slack channel.

srosenberg reviewed (commented):
> Reviewed 1 of 6 files at r3.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/95460)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @dt, @herkolategan, and @stevendanna)

___
*[`pkg/roachprod/roachprod.go` line 441 at r5](https://reviewable.io/reviews/cockroachdb/cockroach/95460#-NMjBqt0uBCnu4KGzgEb-r5-441:-NMoYH3-AGwpxSPfznkm:b-eo406y) ([raw file](https://github.com/cockroachdb/cockroach/blob/3f395ef7a20a8bbe7db7848f95c97d78a3f2747f/pkg/roachprod/roachprod.go#L441)):*
<details><summary><i>Previously, msbutler (Michael Butler) wrote…</i></summary><blockquote>

huh, you're right. The behavior on master is: if one node is specified, ssh onto that node, and pass the cmd to ./cockroach sql. If many nodes are specified, try parallel execution. It's a confusing API, and I think I have over simplified it. I'll change this PR to retain the behavior, and will let test-eng ponder if this API behavior should change.
</blockquote></details>

The issue is that the current API equates `len(c.Nodes) == 1` to interactive mode. This may have made sense when the only consumer of the API was roachprod, but with your changes we've added another consumer, namely roachtest. What about adding an explicit parameter, e.g., `disableInteractiveMode` which would be `true` from roachtest? 

```
if !disableInteractiveMode && (len(args) == 0 || len(c.Nodes) == 1) {
  // open shell
}
// otherwise, RunSQL non-interactively
```


<!-- Sent from Reviewable.io -->

msbutler reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/95460)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @dt, @herkolategan, @renatolabs, @srosenberg, and @stevendanna)

___
*[`pkg/roachprod/roachprod.go` line 441 at r5](https://reviewable.io/reviews/cockroachdb/cockroach/95460#-NMjBqt0uBCnu4KGzgEb-r5-441:-NMofpDMAdKABc7e36yI:b2ck7ov) ([raw file](https://github.com/cockroachdb/cockroach/blob/3f395ef7a20a8bbe7db7848f95c97d78a3f2747f/pkg/roachprod/roachprod.go#L441)):*
<details><summary><i>Previously, srosenberg (Stan Rosenberg) wrote…</i></summary><blockquote>

The issue is that the current API equates `len(c.Nodes) == 1` to interactive mode. This may have made sense when the only consumer of the API was roachprod, but with your changes we've added another consumer, namely roachtest. What about adding an explicit parameter, e.g., `disableInteractiveMode` which would be `true` from roachtest? 

```
if !disableInteractiveMode && (len(args) == 0 || len(c.Nodes) == 1) {
  // open shell
}
// otherwise, RunSQL non-interactively
```
</blockquote></details>

check out my latest push to the first commit. I like my approach because it minimizes the vars you need to keep in mind during this branching. Further, I no longer think `len(args) == 0` should make a difference-- we'd never want to open a sql shell during a multinode roachprod sql cmd. I hope my naming obviates the need for a `disableInteractiveMode` var. 


<!-- Sent from Reviewable.io -->

srosenberg reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/95460)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @dt, @herkolategan, @renatolabs, and @stevendanna)

___
*[`pkg/roachprod/roachprod.go` line 441 at r5](https://reviewable.io/reviews/cockroachdb/cockroach/95460#-NMjBqt0uBCnu4KGzgEb-r5-441:-NMoj1hQ4Dj7Ej3XaS2J:b-anuf54) ([raw file](https://github.com/cockroachdb/cockroach/blob/3f395ef7a20a8bbe7db7848f95c97d78a3f2747f/pkg/roachprod/roachprod.go#L441)):*
<details><summary><i>Previously, msbutler (Michael Butler) wrote…</i></summary><blockquote>

check out my latest push to the first commit. I like my approach because it minimizes the vars you need to keep in mind during this branching. Further, I no longer think `len(args) == 0` should make a difference-- we'd never want to open a sql shell during a multinode roachprod sql cmd. I hope my naming obviates the need for a `disableInteractiveMode` var. 
</blockquote></details>

I think it's fine. However, SQL is still a public method in the API, so nothing technically prevents a roachtest author from calling it by mistake. The comment header is nice; maybe we could emphasize it,

```
// SQL runs `cockroach sql` on a remote cluster. 
//
// NOTE: For a single-node cluster, a pseudo-terminal is attached to ssh which may result in an _interactive_ ssh session.
```


<!-- Sent from Reviewable.io -->

renatolabs reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/95460)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @dt, @herkolategan, @msbutler, @srosenberg, and @stevendanna)

*[a discussion](https://reviewable.io/reviews/cockroachdb/cockroach/95460#-NMoqVsK8K01CPootwuH:-NMoqVsK8K01CPootwuI:bqvlln1) (no related file):*
Maybe we could add a comment to roachtests where we use the `NoBackups()` option to explain that decision (especially non-backup related roachtests). It's not clear to me why we're disabling backups in some cases.

___
___
*[`pkg/roachprod/roachprod.go` line 441 at r5](https://reviewable.io/reviews/cockroachdb/cockroach/95460#-NMjBqt0uBCnu4KGzgEb-r5-441:-NMoq7Np4TdV111lyyXR:bzcjbga) ([raw file](https://github.com/cockroachdb/cockroach/blob/3f395ef7a20a8bbe7db7848f95c97d78a3f2747f/pkg/roachprod/roachprod.go#L441)):*
<details><summary><i>Previously, srosenberg (Stan Rosenberg) wrote…</i></summary><blockquote>

I think it's fine. However, SQL is still a public method in the API, so nothing technically prevents a roachtest author from calling it by mistake. The comment header is nice; maybe we could emphasize it,

```
// SQL runs `cockroach sql` on a remote cluster. 
//
// NOTE: For a single-node cluster, a pseudo-terminal is attached to ssh which may result in an _interactive_ ssh session.
```
</blockquote></details>

I believe this is _always_ opening an interactive shell for 1-node clusters. Maybe the comment should highlight that this function should not be called by roachtest.
___
*[`pkg/roachprod/install/cockroach.go` line 290 at r6](https://reviewable.io/reviews/cockroachdb/cockroach/95460#-NMoooTA90tFQdVNNVXO:-NMoooTBA1gsYVoeudeK:b-s7qnke) ([raw file](https://github.com/cockroachdb/cockroach/blob/1b27c6bd2940e5bcee18e68683dc387d0ea914cc/pkg/roachprod/install/cockroach.go#L290)):*
I'm confused by this comment. `Or` makes it sound like one of two things can happen: `Exec` or `InteractiveSQL`. However, aren't we _always_ opening an interactive SQL shell in this case?
> if no `-e` flag was passed

Is this actually enforced? The only check in the code is regarding number of nodes.
___
*[`pkg/roachprod/install/cockroach.go` line 324 at r6](https://reviewable.io/reviews/cockroachdb/cockroach/95460#-NMoqJcf1nVVn3kIzEQn:-NMoqJcf1nVVn3kIzEQo:bw3cf4c) ([raw file](https://github.com/cockroachdb/cockroach/blob/1b27c6bd2940e5bcee18e68683dc387d0ea914cc/pkg/roachprod/install/cockroach.go#L324)):*
> ```Go
> 		}
> 		cmd += cockroachNodeBinary(c, node) + " sql --url " +
> 			c.NodeURL("localhost", c.NodePort(node), tenantName /* tenantName */) + " " +
> ```

Nit: drop `/* tenantName */`
___
*[`pkg/roachprod/install/cockroach.go` line 794 at r6](https://reviewable.io/reviews/cockroachdb/cockroach/95460#-NMoqNj_3fB4Z1RL-r-F:-NMoqNj_3fB4Z1RL-r-G:bul50g4) ([raw file](https://github.com/cockroachdb/cockroach/blob/1b27c6bd2940e5bcee18e68683dc387d0ea914cc/pkg/roachprod/install/cockroach.go#L794)):*
> ```Go
> CREATE SCHEDULE IF NOT EXISTS test_only_backup FOR BACKUP INTO '%s' %s`,
> 		collectionPath, scheduleArgs)
> 	return c.ExecSQL(ctx, l, "", []string{createScheduleCmd})
> ```

Nit: bring `/* tenantName */` back.


<!-- Sent from Reviewable.io -->

msbutler reviewed (commented):
Will do!

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/95460#-:-NMp78eO2XPoW6bx9I9M:bhy9ost)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @dt, @herkolategan, @renatolabs, @srosenberg, and @stevendanna)

*[a discussion](https://reviewable.io/reviews/cockroachdb/cockroach/95460#-NMoqVsK8K01CPootwuH:-NMp7AjnEbLrSlgreN1N:b-896fix) (no related file):*
<details><summary><i>Previously, renatolabs (Renato Costa) wrote…</i></summary><blockquote>

Maybe we could add a comment to roachtests where we use the `NoBackups()` option to explain that decision (especially non-backup related roachtests). It's not clear to me why we're disabling backups in some cases.
</blockquote></details>

Done.

___
___
*[`pkg/roachprod/roachprod.go` line 441 at r5](https://reviewable.io/reviews/cockroachdb/cockroach/95460#-NMjBqt0uBCnu4KGzgEb-r5-441:-NMp1gCH5u76zn9OsZOE:bnz4awi) ([raw file](https://github.com/cockroachdb/cockroach/blob/3f395ef7a20a8bbe7db7848f95c97d78a3f2747f/pkg/roachprod/roachprod.go#L441)):*
ack. added a bunch of warnings to the comments. 

>> I believe this is always opening an interactive shell for 1-node clusters.

I don't think that's the case if the user passes `-e` to the command? I just checked that this does not start an interactive sql shell on master:
```
roachprod sql local:1 -- -e "SHOW DATABASES";
```
___
*[`pkg/roachprod/install/cockroach.go` line 290 at r6](https://reviewable.io/reviews/cockroachdb/cockroach/95460#-NMoooTA90tFQdVNNVXO:-NMp4I29-ndOHtWcDO89:b-ejrt3d) ([raw file](https://github.com/cockroachdb/cockroach/blob/1b27c6bd2940e5bcee18e68683dc387d0ea914cc/pkg/roachprod/install/cockroach.go#L290)):*
<details><summary><i>Previously, renatolabs (Renato Costa) wrote…</i></summary><blockquote>

I'm confused by this comment. `Or` makes it sound like one of two things can happen: `Exec` or `InteractiveSQL`. However, aren't we _always_ opening an interactive SQL shell in this case?
> if no `-e` flag was passed

Is this actually enforced? The only check in the code is regarding number of nodes.
</blockquote></details>

Maybe we're debating what the word "interactive" means here. My assumption is passing `-e`(or -f) makes the session un-interactive. I've updated the doctstring to hopefully clarify things. I think it would be more confusing to name this funciton `OpenSQLShell` or `InteractiveSQL` as it may imply that the caller cannot pass `-e`. IDK. Maybe all this doesn't matter much, given this function has 1 caller which I introduced :D. 


FWIW, here's what `./cockroach sql -h` says about `-e`:
```
 -e, --execute <stmtlist>
                                      Execute the SQL statement(s) on the command line, then exit. This flag may
                                      be specified multiple times and each value may contain multiple semicolon
                                      separated statements. If an error occurs in any statement, the command exits
                                      with a non-zero status code and further statements are not executed. The
                                      results of each SQL statement are printed on the standard output.  This flag
                                      is incompatible with --file / -f.
```


<!-- Sent from Reviewable.io -->

srosenberg reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/95460)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @dt, @herkolategan, @msbutler, @renatolabs, and @stevendanna)

___
*[`pkg/roachprod/install/cockroach.go` line 290 at r6](https://reviewable.io/reviews/cockroachdb/cockroach/95460#-NMoooTA90tFQdVNNVXO:-NMpTkwS0_9oUefeaHTK:bt67ihf) ([raw file](https://github.com/cockroachdb/cockroach/blob/1b27c6bd2940e5bcee18e68683dc387d0ea914cc/pkg/roachprod/install/cockroach.go#L290)):*
<details><summary><i>Previously, msbutler (Michael Butler) wrote…</i></summary><blockquote>

Maybe we're debating what the word "interactive" means here. My assumption is passing `-e`(or -f) makes the session un-interactive. I've updated the doctstring to hopefully clarify things. I think it would be more confusing to name this funciton `OpenSQLShell` or `InteractiveSQL` as it may imply that the caller cannot pass `-e`. IDK. Maybe all this doesn't matter much, given this function has 1 caller which I introduced :D. 


FWIW, here's what `./cockroach sql -h` says about `-e`:
```
 -e, --execute <stmtlist>
                                      Execute the SQL statement(s) on the command line, then exit. This flag may
                                      be specified multiple times and each value may contain multiple semicolon
                                      separated statements. If an error occurs in any statement, the command exits
                                      with a non-zero status code and further statements are not executed. The
                                      results of each SQL statement are printed on the standard output.  This flag
                                      is incompatible with --file / -f.
```
</blockquote></details>

Right. If `-e` is passed, the query is executed and the stream (and ssh session) is closed; i.e., it's non-interactive, as in the stream is not kept open. Otherwise, the pseudo-terminal (via `ssh -t`) forces a sql shell. Note, without the pseudo-terminal, there is no sql shell, but the stream stays open until it's explicitly terminated. E.g.,

```
ssh -t gce2 go/src/github.com/cockroachdb/cockroach/cockroachshort-linux-2.6.32-gnu-amd64 sql --insecure
#
# Welcome to the CockroachDB SQL shell.
# All statements must be terminated by a semicolon.
# To exit, type: \q.
#
# Server version: CockroachDB CCL v23.1.0-alpha.1-1233-g04f8014761b-dirty (x86_64-unknown-linux-gnu, built 2023/01/25 06:14:20, go1.19.1) (same version as client)
# Cluster ID: 2f913ec5-b420-412f-b2f7-ca4064c07f31
#
# Enter \? for a brief introduction.
#
root@localhost:26257/defaultdb>
```

```
ssh gce2 go/src/github.com/cockroachdb/cockroach/cockroachshort-linux-2.6.32-gnu-amd64 sql --insecure
HELLO
I AM STILL HERE
CONTROL+D
missing semicolon at end of statement: HELLO
ERROR: last statement was not executed: HELLO
Failed running "sql"
```
___
*[`pkg/roachprod/install/cockroach.go` line 291 at r9](https://reviewable.io/reviews/cockroachdb/cockroach/95460#-NMpTayB6SDJoO6R-8Jz:-NMpTayCDkfZef7wimFH:b-540bh3) ([raw file](https://github.com/cockroachdb/cockroach/blob/cbbf62b18e3ea46ceae4cf973afbdf7e91e68269/pkg/roachprod/install/cockroach.go#L291)):*
> ```Go
> // ExecOrInteractiveSQL ssh's onto a single node and executes `./ cockroach sql`
> // with the provided args, potentially opening an interactive session. Note
> // that the caller can pass the `--e` flag to execute sql cmds and exit the
> ```

Nit: the shorthand has a single dash, i.e., `-e`.


<!-- Sent from Reviewable.io -->

renatolabs reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/95460)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @dt, @herkolategan, @msbutler, @srosenberg, and @stevendanna)

___
*[`pkg/roachprod/install/cockroach.go` line 290 at r6](https://reviewable.io/reviews/cockroachdb/cockroach/95460#-NMoooTA90tFQdVNNVXO:-NN2513k2_ZKLAEENsE3:b-ok925d) ([raw file](https://github.com/cockroachdb/cockroach/blob/1b27c6bd2940e5bcee18e68683dc387d0ea914cc/pkg/roachprod/install/cockroach.go#L290)):*
<details><summary><i>Previously, srosenberg (Stan Rosenberg) wrote…</i></summary><blockquote>

Right. If `-e` is passed, the query is executed and the stream (and ssh session) is closed; i.e., it's non-interactive, as in the stream is not kept open. Otherwise, the pseudo-terminal (via `ssh -t`) forces a sql shell. Note, without the pseudo-terminal, there is no sql shell, but the stream stays open until it's explicitly terminated. E.g.,

```
ssh -t gce2 go/src/github.com/cockroachdb/cockroach/cockroachshort-linux-2.6.32-gnu-amd64 sql --insecure
#
# Welcome to the CockroachDB SQL shell.
# All statements must be terminated by a semicolon.
# To exit, type: \q.
#
# Server version: CockroachDB CCL v23.1.0-alpha.1-1233-g04f8014761b-dirty (x86_64-unknown-linux-gnu, built 2023/01/25 06:14:20, go1.19.1) (same version as client)
# Cluster ID: 2f913ec5-b420-412f-b2f7-ca4064c07f31
#
# Enter \? for a brief introduction.
#
root@localhost:26257/defaultdb>
```

```
ssh gce2 go/src/github.com/cockroachdb/cockroach/cockroachshort-linux-2.6.32-gnu-amd64 sql --insecure
HELLO
I AM STILL HERE
CONTROL+D
missing semicolon at end of statement: HELLO
ERROR: last statement was not executed: HELLO
Failed running "sql"
```
</blockquote></details>

I got confused here by the use of the word `Exec` and I was thinking of `syscall.Exec()` given our recent experiences. I understand it now. I still think these APIs are more convoluted than they should be, but this PR is an improvement over what we had before :+1: 


<!-- Sent from Reviewable.io -->

msbutler commented:
Latest [Roachtest nightly run](https://teamcity.cockroachdb.com/viewLog.html?buildId=8480109&buildTypeId=Cockroach_Nightlies_RoachtestNightlyGceBazel) revealed no flakes due to this PR! Posting to eng momentarily..

msbutler commented:
hm, @srosenberg or @renatolabs i think i'll wait to post to eng until after one of y'all have approved the PR. I think i've addressed all your comments, but do let me know if I've missed something!

renatolabs reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/95460#-:-NN3Eisv58WaAGExQywW:bnfp4nl)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @dt, @herkolategan, @msbutler, @srosenberg, and @stevendanna)



<!-- Sent from Reviewable.io -->

srosenberg reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

Thanks for running the sanity checks!

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/95460#-:-NN4HQmSC2SRb5SNQjCo:b-wqmq6n)* status: :shipit: complete! 2 of 0 LGTMs obtained (waiting on @dt, @herkolategan, @msbutler, @renatolabs, and @stevendanna)



<!-- Sent from Reviewable.io -->

msbutler commented:
bors r=renatolabs, srosenberg
